packages/databricks-vscode/src/cli/CliWrapper.test.ts

```diff
@@ -5,62 +5,100 @@ import {
-import {writeFile} from "node:fs/promises";
-
+import {writeFile, readFile} from "node:fs/promises";
+import {when, spy, reset} from "ts-mockito";
 import {CliWrapper} from "./CliWrapper";
 import path from "node:path";
+import os from "node:os";
+import crypto from "node:crypto";
 import {Context} from "@databricks/databricks-sdk/dist/context";
 import {logging} from "@databricks/databricks-sdk";
 
 const execFile = promisify(execFileCb);
+const cliPath = path.join(__dirname, "../../bin/databricks");
+
+function getTempLogFilePath() {
+    return path.join(
+        os.tmpdir(),
+        `databricks-cli-logs-${crypto.randomUUID()}.json`
+    );
+}
+
+function createCliWrapper(logFilePath?: string) {
+    return new CliWrapper(
+        {
+            asAbsolutePath(relativePath: string) {
+                return path.join(__dirname, "../..", relativePath);
+            },
+        } as any,
+        logFilePath
+    );
+}
 
 describe(__filename, () => {
     it("should embed a working databricks CLI", async () => {
-        const cliPath = __dirname + "/../../bin/databricks";
         const result = await execFile(cliPath, ["--help"]);
         assert.ok(result.stdout.indexOf("databricks") > 0);
     });
 
+    let mocks: any[] = [];
+    afterEach(() => {
+        mocks.forEach((mock) => reset(mock));
+        mocks = [];
+    });
+
+    it("should tell CLI to log its output to a file", async () => {
+        const logFilePath = getTempLogFilePath();
+        const configsSpy = spy(workspaceConfigs);
+        mocks.push(configsSpy);
+        when(configsSpy.loggingEnabled).thenReturn(true);
+        const cli = createCliWrapper(logFilePath);
+        await execFile(cli.cliPath, ["version", ...cli.loggingArguments]);
+        const file = await readFile(logFilePath);
+        // Just checking if the file is not empty to avoid depending on internal CLI log patterns
+        assert.ok(file.toString().length > 0);
+    });
+
     it("should create sync commands", async () => {
-        const cli = new CliWrapper({
-            asAbsolutePath(path: string) {
-                return path;
-            },
-        } as any);
+        const logFilePath = getTempLogFilePath();
+        const cli = createCliWrapper(logFilePath);
         const mapper = new SyncDestinationMapper(
-            new LocalUri(
-                Uri.file("/Users/fabian.jakobs/Desktop/notebook-best-practices")
-            ),
+            new LocalUri(Uri.file("/user/project")),
             new RemoteUri(
                 Uri.from({
                     scheme: "wsfs",
-                    path: "/Repos/fabian.jakobs@databricks.com/notebook-best-practices",
+                    path: "/Repos/user@databricks.com/project",
                 })
             )
         );
 
+        const syncCommand = `${cliPath} sync . /Repos/user@databricks.com/project --watch --output json`;
+        const loggingArgs = `--log-level debug --log-file ${logFilePath} --log-format json`;
```

Comment:
You can set these via environment variables. Then you don't need to worry about paths with spaces, etc.

Here's the list: https://github.com/databricks/cli/blob/d70d7445c4f8f873959dd45ae68f8c42617d9353/cmd/root/logger.go#L17

Reply:
I went with the arguments for the consistency sake, as we use them for all other CLI options (including paths for sync destination, which can also contain spaces). As I understand paths with spaces shouldn't be a concern, as nodejs handles normalization for us, plus we don't spawn the process in a shell, so normalising shouldn't be necessary. I haven't tested this on Windows though, will try to double check

Reply:
Understood, I didn't see the array in `loggingArguments` and thought the args were built through string interpolation. Would be great to double check that everything works on Windows.

Reply:
Checked on windows, no problems for paths with spaces
![Screenshot 2023-11-03 at 11 59 14](https://github.com/databricks/databricks-vscode/assets/148094031/4a69613a-c469-4802-b1ef-e62b2cd68bfb)